packages/base/source/adt/union/union.js

```diff
@@ -40,6 +42,29 @@ function mapObject(object, transform) {
   }, {});
 }
 
+//
+// Gets a custom error message for the matchWith function.
+// 
+function getMatchWithErrorMessage(method, property) {
+  return `Variant "${property}" not covered in pattern.
+        This could mean you did not include all variants in your Union's matchWith function.
+
+        For example, if you had this Union:
+
+        const Operation = union({
+            Add: (a, b) => ({ a, b }),
+            Subtract: (a, b) => ({ a, b }),
+        })
+
+        But wrote this matchWith:
+
+        op.matchWith({
+            Add: ({ a, b }) => a + b
+            // Subtract not implemented!
+        })
+
+        It would throw this error because we need to check against 'Subtract'. Check your matchWith function inside of the ${method} method, it's possibly missing the '${property}' in the Object you pass to 'matchWith'.`
```

Comment:
Can you break this paragraph in lines of ~120 characters?

The ` Check your matchWith function inside of the ${method} method, it's possibly missing the '${property}'` part is a bit weird, as you pass the object that contains the methods, rather than a function in particular. I'm not sure if you intended something else with this message, but I think it would be okay to just leave it as `Checkyour matchWith function's argument, it's possibly missing the '${property}' method in the object you've passed'.